packages/editor-ui/src/components/CodeNodeEditor/linter.ts

```diff
@@ -559,6 +559,75 @@ export const useLinter = (
 			});
 		});
 
+		/**
+		 * Lint for `$(variable)` usage where variable is not a string, in both modes.
+		 *
+		 * $(nodeName) -> <no autofix>
+		 */
+		const isDollarSignWithVariable = (node: Node) =>
```

Comment:
Looks like using a template literal needs to be handled separately (we have the same issue in the task runner as well):

<img width="505" alt="image" src="https://github.com/user-attachments/assets/0fd12156-0fef-432d-b47f-1f0732d65388">

Reply:
Good catch, added checks for template string literals with and without vars.